features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/thresholding/FlowThresholding.java

```diff
@@ -0,0 +1,149 @@
+
+        //noinspection unchecked
+        this.sessions = new CacheBuilder<NodeInterfaceKey, Session>()
+                .withConfig(sessionCacheConfig)
+                .withCacheLoader(new CacheLoader<NodeInterfaceKey, Session>() {
+                    @Override
+                    public Session load(final NodeInterfaceKey nik) throws Exception {
+                        final var thresholdingSession = FlowThresholding.this.thresholdingService.createSession(nik.nodeId,
+                                                                                       nik.ifaceAddr,
+                                                                                       SERVICE_NAME,
+                                                                                       FLOW_APP_RRD_REPO,
+                                                                                       new ServiceParameters(Collections.emptyMap()));
+                        final var collectionAgent = FlowThresholding.this.collectionAgentFactory.createCollectionAgent(
+                                Integer.toString(nik.nodeId),
+                                InetAddressUtils.addr(nik.ifaceAddr));
+
+                        return new Session(thresholdingSession,
+                                           collectionAgent);
+                    }
+                }).build();
+    }
+
+    public void threshold(final List<FlowDocument> documents,
+                          final FlowSource source) throws ExecutionException, ThresholdInitializationException {
+
+        for (final var document : documents) {
+            if (document.getNodeExporter() != null && !Strings.isNullOrEmpty(document.getApplication())) {
+                final var nodeId = document.getNodeExporter().getNodeId();
+
+                final var key = new ApplicationKey(nodeId,
+                                                   document.getDirection() == Direction.INGRESS
+                                                   ? document.getInputSnmp()
+                                                   : document.getOutputSnmp(),
+                                                   document.getApplication());
+
+                // Update the counter
+                final var counter = this.applicationAccumulator.computeIfAbsent(key, k -> new AtomicLong(0))
+                                                               .addAndGet(document.getBytes());
+
+                // Apply thresholding
+                final var session = this.sessions.get(new NodeInterfaceKey(nodeId, source.getSourceAddress()));
+
+                final var nodeResource = new NodeLevelResource(nodeId);
+                final var appResource = new DeferredGenericTypeResource(nodeResource, RESOURCE_TYPE_NAME, document.getApplication());
+
+                final var collectionSetBuilder = new CollectionSetBuilder(session.collectionAgent)
+                        .withCounter(appResource, document.getApplication(), "bytes", counter);
```

Comment:
Would it be useful to track `bytesIn` vs `bytesOut` vs `bytesTotal`?

Reply:
We're currently playing around with this. Seems more useful to have a distinct metric per application. This would allow to define thresholds per application.